portfolio/src/main/webapp/script.js

```diff
@@ -28,67 +28,145 @@ function addRandomGreeting() {
 }
 
 function expandDesign(clickedId){
-//whenever you click an image on the description page, this pulls up more images in the category alongside descriptions
-const designPage = document.getElementById('design_container');
-designPage.innerHTML = "";
-console.log(clickedId); //REMOVE WHEN DONE
+    //whenever you click an image on the description page, this pulls up more images in the category alongside descriptions
+    const designPage = document.getElementById('design_container');
+    designPage.innerHTML = "";
 
-//descriptions for every single page that can be clicked on
-const descriptions = 
-    ["Parasite is a fake band created by me and my friends back home."  
-    +"While we have not created any music yet (hence the fake band), "
-    +"I have created plenty of promotional material for us. "
-    +"Hopefully one day the band can become a reality.",
-    "I primarily work in traditional art, as seen by this selection of projects."
-    +"Most of this art was created for my AP Art portfolio, but some pieces"
-    +"have been created since then. Hopefully over this break I can do more"
-    +"traditional art."
-    ]
+    //[title, description]
+    var page_heading = [];
+    var img_ids_desc = [];
```

Comment:
okay, here comes a somewhat lengthy comment. 

I understand what you're going for here with initializing these variable. if you initialize them once, you don't have to do "var" before the variable name when you go to overwrite them below. Additionally, they'll technically exist so, if somehow a user manages to pass an ID that doesn't match any of the IDs you're accounting for in your if/else cases, the lists will exist and things won't crash. 

_However_, things will fall apart when page_heading[0] and page_heading[1] are called. Let's say that my clickedId was "devon" which is not one of the ones you test for. in that case, page_heading will just be an empty list. So calling page_heading[0] will result in an error (because there is nothing inside the list). 

There are a few ways around this. The first is to check that page_heading[0] exists before calling page_heading[0] (so something like if (page_heading.length > 1 { do page_heading[0] and page_heading[1]}) but that might be more complicated than it needs to be. The second option is to avoid using lists at all. Lists are great for when we have an unknown quantity of things or if we want to iterate over a bunch of things. Considering we are always adding a constant number of strings to these lists and never iterating over them, we would be better off just having separate variables for the two strings in each list. For example, here, we can do:

var page_heading = "";
var page_heading_desc = "";

and then just change those four variables explicitly. This way, we are being more descriptive with our naming AND we get the benefit of not crashing if we never change those values (because they're already set to empty strings). 

img_ids_desc is a little more tricky because you DO end up iterating over it, but considering it is a list of lists, the sub lists have the same problem as above. That being said, the way to fix this is to use some sort of named container (object, hash map, struct (i dont think js has structs though... maybe it does) to hold those items in the sub list and then put those named containers in a list and iterate that way, but that's complicated and out of scope here so I would say img_ids_desc is fine as is, especially because you won't crash if nothing is added to it.